### PR TITLE
[KYUUBI #3395] [DOCS] [Subtask] Add PyHive client docs

### LINKED ISSUE #3395: [DOCS] [Subtask] Add PyHive client docs
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Which parts of the documentation do you think need improvement?

Add client docs for PySpark with in `docs/client/python`.

### Affects Version(s)

master

### Improving the documentation

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

close #3395.

Add client docs for PyHive.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: docs/client/python/pyhive.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+For further information about usages and features, eg. DB-API async fetching, using in SQLAlchemy, please refer to [project homepage](https://github.com/dropbox/PyHive).
+
+### DB-API
+
+```python
+from pyhive import hive
+cursor = hive.connect(host=kyuubi_host,port=10009).cursor()
+cursor.execute('SELECT * FROM my_awesome_data LIMIT 10')
+print cursor.fetchone()
+print cursor.fetchall()
```
Comment (pan3793) [inline docs/client/python/pyhive.md:44]: python2 is EOL, let's use python3 syntax in demo code

Comment (bowenliang123) [reply docs/client/python/pyhive.md]: Sure. Fixed.

**Review by pan3793 — APPROVED** [review]

File: docs/client/python/pyhive.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+
+# PyHive
+
+[PyHive](https://github.com/dropbox/PyHive) is a collection of Python DB-API and SQLAlchemy interfaces for Hive. PyHive is able to connect with Kyuubi server as Kyuubi server supports serving in thrift protocol as HiveServer2.
```
Comment (yaooqinn) [inline docs/client/python/pyhive.md:21]: PyHive can connect with the Kyuubi server serving in thrift protocol as HiveServer2.

File: docs/client/python/pyhive.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+PyHive is now marked as [no-longer maintained](https://github.com/dropbox/PyHive/commit/d199a1bd55c656b5c28d0d62f2d3f2e6c9a82a54) and staying on HiveServer2 Thrift bindings v11 which is compatible with Kyuubi server.
```
Comment (yaooqinn) [inline docs/client/python/pyhive.md:23]: this line is unnecessary

File: docs/client/python/pyhive.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+## Requirements
+PyHive works with Python 2.7 / Python 3. Install PyHive via pip for the Hive interface.
+
+```
+pip install 'pyhive[hive]'
+```
+
+## Usage
+Use Kyuubi server's host and thrift protocol port to connect with.
```
Comment (yaooqinn) [inline docs/client/python/pyhive.md:31]: Use **the** Kyuubi server's host and thrift protocol port to connect ~~with~~.

File: docs/client/python/pyhive.md (hunk 3 above)
Comment (bowenliang123) [reply docs/client/python/pyhive.md]: OK, removed. My intention was to warn users who cares about end of support situation and maybe consider any other python lib approach.

**Review by yaooqinn — APPROVED** [review]

File: docs/client/python/pyhive.md (hunk 4 above)
Comment (bowenliang123) [reply docs/client/python/pyhive.md]: ok, applied.

File: docs/client/python/pyhive.md (hunk 2 above)
Comment (bowenliang123) [reply docs/client/python/pyhive.md]: ok

File: docs/client/python/pyhive.md (hunk 3 above)
Comment (yaooqinn) [reply docs/client/python/pyhive.md]: > My intention was to warn users who cares about end of support situation

the status of a third-party project is always unforeseeable, we can avoid making statements for them.

> any other python lib approach.

I hope its name will be pykyuubi:)

**Comment by bowenliang123** [comment]
Force pushed squashed commit and ready for the merge.

**Comment by yaooqinn** [comment]
thanks, merged to master

File: docs/client/python/pyhive.md (hunk 3 above)
Comment (bowenliang123) [reply docs/client/python/pyhive.md]: Got your point. Avoid comments on other project. I have one sec hesitating to whether fill this PR as I found out PyHive is not maintained anymore. But after all , it's handy to use and approach to kyuubi source.

As for `pykyuubi`, I think we will have one in far future as we introduce more and more protocols and FE (like FlighSQL) libs for different usages. It must provide unique improvement other than just wrap some libs in it. Looking forward to it. 😄

**Comment by bowenliang123** [comment]
@yaooqinn Is it possible to push this pyHive client docs to v1.6.0 as well?

**Pull request closed without merging**